File: LeetCode_Solutions/409-longest-palindrome/longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        ans=0
        cnt=collections.Counter(s)
        
        for element in cnt.values():
            
            ans+=element if element%2==0 else element-1
            
        
        oddCount=any(c%2==1 for c in cnt.values())
        
        ans+=oddCount
        return ans

File: LeetCode_Solutions/409-longest-palindrome/longest-palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        Map<Character,Integer> mp=new HashMap<>();
        int[] cnt=new int[128];
        int ans=0;
        for(char c:s.toCharArray()){
            ++cnt[c];
        }
        for(int freq:cnt){
            ans+=freq%2==0?freq:freq-1;
        }
        boolean oddCnt=Arrays.stream(cnt).anyMatch(freq->freq%2==1);
        ans+=oddCnt?1:0;
        // for(int c:cnt){
        //     System.out.println(c);
        // }
        return ans;
    }
}